src/components/DistanceRequest.js

```diff
@@ -1,40 +1,145 @@
-import React, {useEffect} from 'react';
-import {View} from 'react-native';
+import React, {useEffect, useState} from 'react';
+import {ScrollView, View} from 'react-native';
+import lodashGet from 'lodash/get';
+import _ from 'underscore';
 import PropTypes from 'prop-types';
 import {withOnyx} from 'react-native-onyx';
-import Text from './Text';
-import * as IOU from '../libs/actions/IOU';
-import styles from '../styles/styles';
 import ONYXKEYS from '../ONYXKEYS';
+import * as Transaction from '../libs/actions/Transaction';
+import MenuItemWithTopDescription from './MenuItemWithTopDescription';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import compose from '../libs/compose';
+import * as Expensicons from './Icon/Expensicons';
+import theme from '../styles/themes/default';
+import Button from './Button';
+import styles from '../styles/styles';
+import LinearGradient from './LinearGradient';
+
+const MAX_WAYPOINTS = 25;
+const MAX_WAYPOINTS_TO_DISPLAY = 4;
 
 const propTypes = {
     /** The transactionID of this request */
     transactionID: PropTypes.string,
+
+    /** The optimistic transaction for this request */
+    transaction: PropTypes.shape({
+        transactionID: PropTypes.string,
+        comment: PropTypes.shape({
+            waypoints: PropTypes.shape({
+                lat: PropTypes.number,
+                lng: PropTypes.number,
+                address: PropTypes.string,
+            }),
+        }),
+    }),
+
+    ...withLocalizePropTypes,
 };
 
 const defaultProps = {
     transactionID: '',
+    transaction: {},
 };
```

Comment:
Default props can be set while destructing props
https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#destructuring

Reply:
I tried that but then eslint complained. Many components still use default props so I think we can keep this for now. cc @hayata-suenaga since you wrote this, what do you think is happening here.
![image](https://github.com/Expensify/App/assets/26260477/4f6bc45b-9509-4ecd-b389-ace6862e6929)

Reply:
I thought we returned down that ESLint rule

@neil-marcellini what is the exact ESLint rule that is warning?